instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -48,37 +54,78 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, M
                   s.record_exception(reply)
                   s.status = Trace::Status.error(reply.message)
                 end
+                next unless config[:record_value_size] && commands.length == 1
+
+                if RETRIEVED_VALUE_SIZE_COMMANDS.include?(commands[0][0])
+                  s['db.retrieved_value_size_bytes'] = calculate_bytesize(reply)
+                end
               end
             end
           end
 
           private
 
+          def calculate_bytesize(value) # rubocop:disable Metrics/CyclomaticComplexity
+            case value
+            when ::Redis::CommandError
+              # Don't report on these; call it zero.
+              0
+            when String
+              value.bytesize
+            when Array
+              value.sum { |s| calculate_bytesize(s) }
+            when Integer
+              value.size
+            when Float
+              # Redis stores floats as strings, so get bytesize of stringified
+              # float
+              value.to_s.bytesize
```

Comment:
I'm always sad when we have to serialize something to calculate its size in instrumentation, and then throw away the serialized representation. I don't suppose there's a way to estimate the size without actually allocating a string and serializing the float?

Is there any difference in the output of `.to_s` vs the wire representation? E.g. rounding or truncation?

Reply:
🤔 this is a great question - I will 👀